### ISSUE TITLE
Allow getting connection pools for an event loop in clients

### ISSUE THREAD

**Issue by wfltaylor** [vapor/redis#201]
**Is your feature request related to a problem? Please describe.**
It would be useful to be able to get the `RedisConnectionPool` for an `EventLoop`.  For example, one may want to extend `QueueContext` with a Redis property, to allow using Redis from Vapor Jobs.

**Describe the solution you'd like**
The method `pool(eventLoop:)` in `Application.Redis` could be made public.

**Describe alternatives you've considered**
This library could extend each type that could make use of Redis.

**Comment by Mordil** [comment]
@wfltaylor Do you have a concrete example use case of what you’re trying to do in which the current API doesn’t allow?

Most of the API surface exposes limited access for a reason - either abstraction from a specific database engine, or to limit the chance of having bad state.

**Comment by Mordil** [comment]
@wfltaylor Do you have a concrete example use case of what you’re trying to do in which the current API doesn’t allow?

Most of the API surface exposes limited access for a reason - either abstraction from a specific database engine, or to limit the chance of having bad state.

**Comment by wfltaylor** [comment]
I am trying to extend 'QueueContext' from Vapor Queues with a `redis` property that uses the correct event loop.

**Comment by Mordil** [comment]
To do what exactly? To have access to Redis for executing commands as part of the job’s process?

This might be more of a feature request for Queues more specifically

**Comment by wfltaylor** [comment]
> To have access to Redis for executing commands as part of the job’s process?

Yes exactly - I need to be able to read and write to Redis during the execution of the job.
